Config.org
#+title: DWM Configuration
#+PROPERTY: header-args:c :tangle ./config.h :mkdirp yes
#+begin_src c
  /* See LICENSE file for copyright and license details. */

  /* appearance */
  static const unsigned int borderpx  = 3;        /* border pixel of windows */
  static const unsigned int snap      = 32;       /* snap pixel */ 
  static const int swallowfloating    = 0;        /* 1 means swallow floating windows by default */ 
  static const unsigned int gappih    = 20;       /* horiz inner gap between windows */ 
  static const unsigned int gappiv    = 10;       /* vert inner gap between windows */
  static const unsigned int gappoh    = 10;       /* horiz outer gap between windows and screen edge */
  static const unsigned int gappov    = 30;       /* vert outer gap between windows and screen edge */
  static       int smartgaps          = 0;        /* 1 means no outer gap when there is only one window */
  static const int showbar            = 1;        /* 0 means no bar */
  static const int topbar             = 1;        /* 0 means bottom bar */
  static const char *fonts[]          = { "JetBrains Mono:size=10", "JoyPixels:size=10" };
  static const char dmenufont[]       = "JetBrains Mono:size=10";
  static const char col_gray1[]       = "#222222";
  static const char col_gray2[]       = "#111111"; static const char col_gray3[]       = "#bbbbbb";
  static const char col_gray4[]       = "#eeeeee";
  static const char col_cyan[]        = "#111111";
  static const char *colors[][3]      = {
	  /*               fg         bg         border   */
	  [SchemeNorm] = { col_gray3, col_gray1, col_gray2 },
	  [SchemeSel]  = { col_gray4, col_cyan,  col_cyan  },
  };

#+end_src
#+begin_src c
  /* tagging */
  static const char *tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };

  static const Rule rules[] = {
	  /* xprop(1):
	   ,*	WM_CLASS(STRING) = instance, class
	   ,*	WM_NAME(STRING) = title
	   ,*/
	  /* class      instance    title       tags mask     isfloating   monitor */
	  { "Gimp",     NULL,       NULL,       0,            1,           -1 },
	  { "Firefox",  NULL,       NULL,       1 << 8,       0,           -1 },

	  { "St",      NULL,     NULL,           0,         0,          1,           0,        -1 },
	  { NULL,      NULL,     "Event Tester", 0,         0,          0,           1,        -1 }, /* xev */
  };

#+end_src
#+begin_src c
  /* layout(s) */
  static const float mfact     = 0.55; /* factor of master area size [0.05..0.95] */
  static const int nmaster     = 1;    /* number of clients in master area */
  static const int resizehints = 1;    /* 1 means respect size hints in tiled resizals */

  #define FORCE_VSPLIT 1  /* nrowgrid layout: force two clients to always split vertically */
  #include "vanitygaps.c"

  static const Layout layouts[] = {
	  /* symbol     arrange function */
	  { "[]=",      tile },    /* first entry is default */
	  { "[M]",      monocle },
	  { "[@]",      spiral },
	  { "[\\]",     dwindle },
	  { "H[]",      deck },
	  { "TTT",      bstack },
	  { "===",      bstackhoriz },
	  { "HHH",      grid },
	  { "###",      nrowgrid },
	  { "---",      horizgrid },
	  { ":::",      gaplessgrid },
	  { "|M|",      centeredmaster },
	  { ">M>",      centeredfloatingmaster },
	  { "><>",      NULL },    /* no layout function means floating behavior */
	  { NULL,       NULL },
  };

#+end_src
#+begin_src c
  /* key definitions */
  #define MODKEY Mod4Mask
  #define TAGKEYS(KEY,TAG) \
	  { MODKEY,                       KEY,      view,           {.ui = 1 << TAG} }, \
	  { MODKEY|ControlMask,           KEY,      toggleview,     {.ui = 1 << TAG} }, \
	  { MODKEY|ShiftMask,             KEY,      tag,            {.ui = 1 << TAG} }, \
	  { MODKEY|ControlMask|ShiftMask, KEY,      toggletag,      {.ui = 1 << TAG} },

  /* helper for spawning shell commands in the pre dwm-5.0 fashion */
  #define SHCMD(cmd) { .v = (const char*[]){ "/bin/sh", "-c", cmd, NULL } }

#+end_src
#+begin_src c
  /* commands */
  static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
  static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
  static const char *termcmd[]  = { "st", NULL };
  static const char scratchpadname[] = "scratchpad";
  static const char *scratchpadcmd[] = { "st", "-t", scratchpadname, "-g", "120x34", NULL };

#+end_src
#+begin_src c
  static Key keys[] = {
	  /* modifier                     key        function        argument */
	  { MODKEY,                       XK_d,      spawn,          {.v = dmenucmd } },
	  { MODKEY,             XK_Return, spawn,          {.v = termcmd } },
	  { MODKEY,                       XK_grave,  togglescratch,  {.v = scratchpadcmd } },
	  { MODKEY,                       XK_b,      togglebar,      {0} },
	  { MODKEY,                       XK_j,      focusstack,     {.i = +1 } },
	  { MODKEY,                       XK_k,      focusstack,     {.i = -1 } },
	  { MODKEY,                       XK_i,      incnmaster,     {.i = +1 } },
	  { MODKEY,                       XK_Delete,      incnmaster,     {.i = -1 } },
	  { MODKEY,                       XK_h,      setmfact,       {.f = -0.05} },
	  { MODKEY,                       XK_l,      setmfact,       {.f = +0.05} },
	  { MODKEY|ShiftMask,             XK_Return, zoom,           {0} },
	  { MODKEY,						XK_z,      incrgaps,       {.i = +3 } },
	  { MODKEY,						XK_x,      incrgaps,       {.i = -3 } },
	  //{ MODKEY|Mod1Mask,              XK_i,      incrigaps,      {.i = +1 } },
	  //{ MODKEY|Mod1Mask|ShiftMask,    XK_i,      incrigaps,      {.i = -1 } },
	  //{ MODKEY|Mod1Mask,              XK_o,      incrogaps,      {.i = +1 } },
	  //{ MODKEY|Mod1Mask|ShiftMask,    XK_o,      incrogaps,      {.i = -1 } },
	  //{ MODKEY|Mod1Mask,              XK_6,      incrihgaps,     {.i = +1 } },
	  //{ MODKEY|Mod1Mask|ShiftMask,    XK_6,      incrihgaps,     {.i = -1 } },
	  //{ MODKEY|Mod1Mask,              XK_7,      incrivgaps,     {.i = +1 } },
	  //{ MODKEY|Mod1Mask|ShiftMask,    XK_7,      incrivgaps,     {.i = -1 } },
	  //{ MODKEY|Mod1Mask,              XK_8,      incrohgaps,     {.i = +1 } },
	  //{ MODKEY|Mod1Mask|ShiftMask,    XK_8,      incrohgaps,     {.i = -1 } },
	  //{ MODKEY|Mod1Mask,              XK_9,      incrovgaps,     {.i = +1 } },
	  //{ MODKEY|Mod1Mask|ShiftMask,    XK_9,      incrovgaps,     {.i = -1 } },
	  { MODKEY,						XK_a,      togglegaps,     {0} },
	  { MODKEY|ShiftMask,				XK_a,      defaultgaps,    {0} },
	  { MODKEY,                       XK_Tab,    view,           {0} },
	  { MODKEY,						XK_q,      killclient,     {0} },
	  { MODKEY,                       XK_t,      setlayout,      {.v = &layouts[0]} },
	  { MODKEY,                       XK_f,      setlayout,      {.v = &layouts[1]} },
	  { MODKEY,                       XK_m,      setlayout,      {.v = &layouts[2]} },
	  { MODKEY|ShiftMask,             XK_f,      fullscreen,     {0} },
	  { MODKEY|ControlMask,			XK_comma,  cyclelayout,    {.i = -1 } },
	  { MODKEY|ControlMask,           XK_period, cyclelayout,    {.i = +1 } },
	  { MODKEY,                       XK_space,  setlayout,      {0} },
	  { MODKEY|ShiftMask,             XK_space,  togglefloating, {0} },
	  { MODKEY,                       XK_s,      togglesticky,   {0} },
	  { MODKEY,                       XK_0,      view,           {.ui = ~0 } },
	  { MODKEY|ShiftMask,             XK_0,      tag,            {.ui = ~0 } },
	  { MODKEY,                       XK_comma,  focusmon,       {.i = -1 } },
	  { MODKEY,                       XK_period, focusmon,       {.i = +1 } },
	  { MODKEY|ShiftMask,             XK_comma,  tagmon,         {.i = -1 } },
	  { MODKEY|ShiftMask,             XK_period, tagmon,         {.i = +1 } },
	  { MODKEY,						XK_w,	   spawn,		   SHCMD("brave") },
	  { MODKEY|ShiftMask,				XK_w,	   spawn,		   SHCMD("st -e nmtui") },
	  { MODKEY,						XK_r,	   spawn,		   SHCMD("st -e lf") },
	  { MODKEY|ShiftMask,				XK_r,	   spawn,		   SHCMD("st -e top") },
	  { MODKEY,						XK_F4,	   spawn,		   SHCMD("st -e pulsemixer") },
	  { MODKEY,						XK_minus,  spawn,		   SHCMD("pulsemixer --change-volume -3") }, 
	  { MODKEY|ShiftMask,				XK_minus,  spawn,		   SHCMD("pulsemixer --change-volume -10") }, 
	  { MODKEY,						XK_equal,  spawn,		   SHCMD("pulsemixer --change-volume +3") }, 
	  { MODKEY|ShiftMask,				XK_equal,  spawn,		   SHCMD("pulsemixer --change-volume +10") }, 
	  { MODKEY|Mod1Mask,				XK_v,	   spawn,		   SHCMD("/home/hacker/fl/scripts/vidplay") },
	  { MODKEY,						XK_BackSpace,    spawn,		   SHCMD("/home/hacker/fl/scripts/shutpc") },
	  { MODKEY|ControlMask,						XK_w,    spawn,		   SHCMD("/home/hacker/fl/scripts/cg.sh") },


	  // Music stuff
	  { MODKEY|ShiftMask,				XK_m,				spawn,		   SHCMD("st -e ncmpcpp") },
	  { MODKEY,						XK_p,				spawn,		   SHCMD("mpc toggle") },
	  { MODKEY|ControlMask,			XK_p,				spawn,		   SHCMD("playerctl play-pause") },
	  { MODKEY|ShiftMask,				XK_p,				spawn,		SHCMD("mpc pause ; pauseallmpv") },
	  { MODKEY,						XK_bracketleft,		spawn,		SHCMD("mpc seek -10") },
	  { MODKEY|ShiftMask,				XK_bracketleft,		spawn,		SHCMD("mpc seek -60") },
	  { MODKEY,						XK_bracketright,	spawn,		SHCMD("mpc seek +10") },
	  { MODKEY|ShiftMask,				XK_bracketright,	spawn,		SHCMD("mpc seek +60") },
	  { MODKEY,						XK_comma,	spawn,		SHCMD("mpc prev") },
	  { MODKEY|ShiftMask,				XK_comma,	spawn,		SHCMD("mpc seek 0%") },
	  { MODKEY,						XK_period,	spawn,		SHCMD("mpc next") },
	  { MODKEY|ShiftMask,				XK_period,	spawn,		SHCMD("mpc repeat") },
	  { 0,							XK_Print,  spawn,		   SHCMD("maim full-$(date '+%y%m%d-%H%M-%S').png") },
	  { ShiftMask,					XK_Print,  spawn,		   SHCMD("/home/hacker/fl/scripts/dmenurecshot") },
	  { MODKEY|ControlMask,			XK_Print, spawn,		   SHCMD("/home/hacker/fl/scripts/dmenurecord") },
	  TAGKEYS(                        XK_1,                      0)
	  TAGKEYS(                        XK_2,                      1)
	  TAGKEYS(                        XK_3,                      2)
	  TAGKEYS(                        XK_4,                      3)
	  TAGKEYS(                        XK_5,                      4)
	  TAGKEYS(                        XK_6,                      5)
	  TAGKEYS(                        XK_7,                      6)
	  TAGKEYS(                        XK_8,                      7)
	  TAGKEYS(                        XK_9,                      8)
	  { MODKEY|ShiftMask,             XK_q,      quit,           {0} },
  };

#+end_src
#+begin_src c
  /* button definitions */
  /* click can be ClkTagBar, ClkLtSymbol, ClkStatusText, ClkWinTitle, ClkClientWin, or ClkRootWin */
  static Button buttons[] = {
	  /* click                event mask      button          function        argument */
	  { ClkLtSymbol,          0,              Button1,        setlayout,      {0} },
	  { ClkLtSymbol,          0,              Button3,        setlayout,      {.v = &layouts[2]} },
	  { ClkWinTitle,          0,              Button2,        zoom,           {0} },
	  { ClkStatusText,        0,              Button2,        spawn,          {.v = termcmd } },
	  { ClkClientWin,         MODKEY,         Button1,        movemouse,      {0} },
	  { ClkClientWin,         MODKEY,         Button2,        togglefloating, {0} },
	  { ClkClientWin,         MODKEY,         Button3,        resizemouse,    {0} },
	  { ClkTagBar,            0,              Button1,        view,           {0} },
	  { ClkTagBar,            0,              Button3,        toggleview,     {0} },
	  { ClkTagBar,            MODKEY,         Button1,        tag,            {0} },
	  { ClkTagBar,            MODKEY,         Button3,        toggletag,      {0} },
  };

#+end_src


config.h
/* See LICENSE file for copyright and license details. */

/* appearance */
static const unsigned int borderpx  = 3;        /* border pixel of windows */

static const unsigned int snap      = 32;       /* snap pixel */ 
static const int swallowfloating    = 0;        /* 1 means swallow floating windows by default */ 
static const unsigned int gappih    = 20;       /* horiz inner gap between windows */ 
static const unsigned int gappiv    = 10;       /* vert inner gap between windows */
static const unsigned int gappoh    = 10;       /* horiz outer gap between windows and screen edge */
static const unsigned int gappov    = 30;       /* vert outer gap between windows and screen edge */
static       int smartgaps          = 0;        /* 1 means no outer gap when there is only one window */
static const int showbar            = 1;        /* 0 means no bar */
static const int topbar             = 1;        /* 0 means bottom bar */
static const char *fonts[]          = { "Cascadia Code:size=10", "JoyPixels:size=10" };
static const char dmenufont[]       = "Cascadia Code:size=10";
static const char col_gray1[]       = "#222222";
static const char col_gray2[]       = "#111111"; static const char col_gray3[]       = "#bbbbbb";
static const char col_gray4[]       = "#eeeeee";
static const char col_cyan[]        = "#111111";
static const char *colors[][3]      = {
	/*               fg         bg         border   */
	[SchemeNorm] = { col_gray3, col_gray1, col_gray2 },
	[SchemeSel]  = { col_gray4, col_cyan,  col_cyan  },
};

/* tagging */
static const char *tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };

static const Rule rules[] = {
	/* xprop(1):
	 *	WM_CLASS(STRING) = instance, class
	 *	WM_NAME(STRING) = title
	 */
	/* class      instance    title       tags mask     isfloating   monitor */
	{ "Gimp",     NULL,       NULL,       0,            1,           -1 },
	{ "Firefox",  NULL,       NULL,       1 << 8,       0,           -1 },

	{ "St",      NULL,     NULL,           0,         0,          1,           0,        -1 },
	{ NULL,      NULL,     "Event Tester", 0,         0,          0,           1,        -1 }, /* xev */
};

/* layout(s) */
static const float mfact     = 0.55; /* factor of master area size [0.05..0.95] */
static const int nmaster     = 1;    /* number of clients in master area */
static const int resizehints = 1;    /* 1 means respect size hints in tiled resizals */

#define FORCE_VSPLIT 1  /* nrowgrid layout: force two clients to always split vertically */
#include "vanitygaps.c"

static const Layout layouts[] = {
	/* symbol     arrange function */
	{ "[]=",      tile },    /* first entry is default */
	{ "[M]",      monocle },
	{ "[@]",      spiral },
	{ "[\\]",     dwindle },
	{ "H[]",      deck },
	{ "TTT",      bstack },
	{ "===",      bstackhoriz },
	{ "HHH",      grid },
	{ "###",      nrowgrid },
	{ "---",      horizgrid },
	{ ":::",      gaplessgrid },
	{ "|M|",      centeredmaster },
	{ ">M>",      centeredfloatingmaster },
	{ "><>",      NULL },    /* no layout function means floating behavior */
	{ NULL,       NULL },
};

/* key definitions */
#define MODKEY Mod4Mask
#define TAGKEYS(KEY,TAG) \
	{ MODKEY,                       KEY,      view,           {.ui = 1 << TAG} }, \
	{ MODKEY|ControlMask,           KEY,      toggleview,     {.ui = 1 << TAG} }, \
	{ MODKEY|ShiftMask,             KEY,      tag,            {.ui = 1 << TAG} }, \
	{ MODKEY|ControlMask|ShiftMask, KEY,      toggletag,      {.ui = 1 << TAG} },

/* helper for spawning shell commands in the pre dwm-5.0 fashion */
#define SHCMD(cmd) { .v = (const char*[]){ "/bin/sh", "-c", cmd, NULL } }

/* commands */
static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-l", "20", "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
static const char *termcmd[]  = { "st", NULL };
static const char scratchpadname[] = "scratchpad";
static const char *scratchpadcmd[] = { "st", "-t", scratchpadname, "-g", "120x34", NULL };

// START_KEYS

static Key keys[] = {
	/* modifier                     key        function        argument */
	//{ MODKEY,						XK_w,	   spawn,		   SHCMD("dmenu_run -l 20") },
	{ MODKEY,                       XK_d,      spawn,          {.v = dmenucmd } },
	{ MODKEY,						XK_Return, spawn,          {.v = termcmd } },
	{ MODKEY,                       XK_grave,  togglescratch,  {.v = scratchpadcmd } },
	{ MODKEY,                       XK_b,      togglebar,      {0} },
	{ MODKEY,                       XK_j,      focusstack,     {.i = +1 } },
	{ MODKEY,                       XK_k,      focusstack,     {.i = -1 } },
	{ MODKEY,                       XK_i,      incnmaster,     {.i = +1 } },
	{ MODKEY,                       XK_Delete,      incnmaster,     {.i = -1 } },
	{ MODKEY,                       XK_h,      setmfact,       {.f = -0.05} },
	{ MODKEY,                       XK_l,      setmfact,       {.f = +0.05} },
	{ MODKEY|ShiftMask,             XK_Return, zoom,           {0} },
	{ MODKEY,						XK_z,      incrgaps,       {.i = +3 } },
	{ MODKEY,						XK_x,      incrgaps,       {.i = -3 } },
	{ MODKEY,						XK_a,      togglegaps,     {0} },
	{ MODKEY|ShiftMask,				XK_a,      defaultgaps,    {0} },
	{ MODKEY,                       XK_Tab,    view,           {0} },
	{ MODKEY,						XK_q,      killclient,     {0} },
	{ MODKEY,                       XK_t,      setlayout,      {.v = &layouts[0]} },
	{ MODKEY,                       XK_f,      setlayout,      {.v = &layouts[1]} },
	{ MODKEY,                       XK_m,      setlayout,      {.v = &layouts[2]} },
	{ MODKEY|ShiftMask,             XK_f,      fullscreen,     {0} },
	{ MODKEY|ControlMask,			XK_comma,  cyclelayout,    {.i = -1 } },
	{ MODKEY|ControlMask,           XK_period, cyclelayout,    {.i = +1 } },
	{ MODKEY,                       XK_space,  setlayout,      {0} },
	{ MODKEY|ShiftMask,             XK_space,  togglefloating, {0} },
	{ MODKEY,                       XK_s,      togglesticky,   {0} },
	{ MODKEY,                       XK_0,      view,           {.ui = ~0 } },
	{ MODKEY|ShiftMask,             XK_0,      tag,            {.ui = ~0 } },
	{ MODKEY,                       XK_comma,  focusmon,       {.i = -1 } },
	{ MODKEY,                       XK_period, focusmon,       {.i = +1 } },
	{ MODKEY|ShiftMask,             XK_comma,  tagmon,         {.i = -1 } },
	{ MODKEY|ShiftMask,             XK_period, tagmon,         {.i = +1 } },
	{ MODKEY,						XK_w,	   spawn,		   SHCMD("librewolf") },
	{ MODKEY|ShiftMask,				XK_t,	   spawn,		   SHCMD("/home/hacker/fl/scripts/tmux-session") },
	{ MODKEY|ShiftMask,				XK_w,	   spawn,		   SHCMD("st -e nmtui") },
	{ MODKEY,						XK_r,	   spawn,		   SHCMD("st -e lf") },
	{ MODKEY|ShiftMask,				XK_r,	   spawn,		   SHCMD("st -e top") },
	{ MODKEY,						XK_F4,	   spawn,		   SHCMD("st -e pulsemixer") },
	{ MODKEY,						XK_minus,  spawn,		   SHCMD("pulsemixer --change-volume -3") }, 
	{ MODKEY|ShiftMask,				XK_minus,  spawn,		   SHCMD("pulsemixer --change-volume -10") }, 
	{ MODKEY,						XK_equal,  spawn,		   SHCMD("pulsemixer --change-volume +3") }, 
	{ MODKEY|ShiftMask,				XK_equal,  spawn,		   SHCMD("pulsemixer --change-volume +10") }, 
	{ MODKEY|Mod1Mask,				XK_v,	   spawn,		   SHCMD("/home/hacker/fl/scripts/vidplay") },
	{ MODKEY,						XK_BackSpace,   spawn,		   SHCMD("/home/hacker/fl/scripts/shutpc") },
	{ MODKEY|ControlMask,			XK_w,			spawn,		   SHCMD("/home/hacker/fl/scripts/cg.sh") },
	{ MODKEY|Mod1Mask,			XK_p,			spawn,		   SHCMD("st -e python") },


	// Music stuff
	{ MODKEY|ShiftMask,				XK_m,				spawn,			SHCMD("st -e ncmpcpp") },
	{ MODKEY,						XK_p,				spawn,			SHCMD("mpc toggle") },
	{ MODKEY|ControlMask,			XK_p,				spawn,			SHCMD("playerctl play-pause") },
	{ MODKEY|ShiftMask,				XK_p,				spawn,			SHCMD("mpc pause ; pauseallmpv") },
	{ MODKEY,						XK_bracketleft,		spawn,			SHCMD("mpc seek -10") },
	{ MODKEY|ShiftMask,				XK_bracketleft,		spawn,			SHCMD("mpc seek -60") },
	{ MODKEY,						XK_bracketright,	spawn,			SHCMD("mpc seek +10") },
	{ MODKEY|ShiftMask,				XK_bracketright,	spawn,			SHCMD("mpc seek +60") },
	{ MODKEY,						XK_comma,			spawn,			SHCMD("mpc prev") },
	{ MODKEY|ShiftMask,				XK_comma,			spawn,			SHCMD("mpc seek 0%") },
	{ MODKEY,						XK_period,			spawn,			SHCMD("mpc next") },
	{ MODKEY|ShiftMask,				XK_period,			spawn,			SHCMD("mpc repeat") },
	{ 0,							XK_Print,			spawn,			SHCMD("maim full-$(date '+%y%m%d-%H%M-%S').png") },
	{ ShiftMask,					XK_Print,			spawn,			SHCMD("/home/hacker/fl/scripts/dmenurecshot") },
	{ MODKEY|ControlMask,			XK_Print,			spawn,			SHCMD("/home/hacker/fl/scripts/dmenurecord") },
	TAGKEYS(                        XK_1,                      0)
	TAGKEYS(                        XK_2,                      1)
	TAGKEYS(                        XK_3,                      2)
	TAGKEYS(                        XK_4,                      3)
	TAGKEYS(                        XK_5,                      4)
	TAGKEYS(                        XK_6,                      5)
	TAGKEYS(                        XK_7,                      6)
	TAGKEYS(                        XK_8,                      7)
	TAGKEYS(                        XK_9,                      8)
	{ MODKEY|ShiftMask,             XK_q,      quit,           {0} },
};

// END_KEYS
/* button definitions */
/* click can be ClkTagBar, ClkLtSymbol, ClkStatusText, ClkWinTitle, ClkClientWin, or ClkRootWin */
static Button buttons[] = {
	/* click                event mask      button          function        argument */
	{ ClkLtSymbol,          0,              Button1,        setlayout,      {0} },
	{ ClkLtSymbol,          0,              Button3,        setlayout,      {.v = &layouts[2]} },
	{ ClkWinTitle,          0,              Button2,        zoom,           {0} },
	{ ClkStatusText,        0,              Button2,        spawn,          {.v = termcmd } },
	{ ClkClientWin,         MODKEY,         Button1,        movemouse,      {0} },
	{ ClkClientWin,         MODKEY,         Button2,        togglefloating, {0} },
	{ ClkClientWin,         MODKEY,         Button3,        resizemouse,    {0} },
	{ ClkTagBar,            0,              Button1,        view,           {0} },
	{ ClkTagBar,            0,              Button3,        toggleview,     {0} },
	{ ClkTagBar,            MODKEY,         Button1,        tag,            {0} },
	{ ClkTagBar,            MODKEY,         Button3,        toggletag,      {0} },
};
